lib/resourcemerge/batch.go

```diff
@@ -3,26 +3,34 @@ package resourcemerge
 import (
 	batchv1 "k8s.io/api/batch/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/klog"
 )
 
 // EnsureJob ensures that the existing matches the required.
 // modified is set to true when existing had to be updated with required.
 func EnsureJob(modified *bool, existing *batchv1.Job, required batchv1.Job) {
-	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
+	// temp trace logging
+	klog.Info("batch.EnsureJob 1")
 
-	if existing.Spec.Selector == nil {
-		*modified = true
-		existing.Spec.Selector = required.Spec.Selector
-	}
-	if !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
-		*modified = true
-		existing.Spec.Selector = required.Spec.Selector
+	// To set Selector you must set ManualSelector to True. If 'required'
+	// sets Selector to value not equal to existing, let current Job
+	// finish as Selector is immutable.
+	if required.Spec.ManualSelector != nil {
+
+		// If ManualSelector is not True we don't care about Selector
+		if required.Spec.ManualSelector {
+			if !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
+				return
+			}
+		}
+		setBoolPtr(modified, &existing.Spec.ManualSelector, required.Spec.ManualSelector)
 	}
+	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
+
```

Comment:
@wking check my re-factoring logic here per my comments/code changes

Reply:
Assorted comments inline, no need for a separate thread on them all ;).